vcd/resource_vcd_vapp_vm.go

```diff
@@ -1238,9 +1238,14 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}, ex
 		}
 
 		if memoryNeedsColdChange || executionType == "create" {
-			err = vm.ChangeMemory(int64(d.Get("memory").(int)))
-			if err != nil {
-				return err
+			memory, isMemorySet := d.GetOk("memory")
+			isMemoryComingFromSizingPolicy := computePolicy != nil && (computePolicy.Memory != nil && !isMemorySet)
```

Comment:
I'm reading the last part `&& !isMemorySet)`. Please confirm that if you do have `memory` set in VM, it works and takes precedence over the one in policy. Thanks!

Reply:
I'm reviewing the entire PR as this unveiled some extra use cases, as I only checked a Sizing Policy that contained all attributes set (cpu, cpu cores and memory), but when an attribute is missing, one must populate the corresponding attribute in the VM resource.

I'll add these new ones and re-test.

Reply:
Did VCD throw some error in that case? The reason I'm asking is because we try to keep the code lean in these cases. Having too much logic at our level usually bites us back in the long term (as we may not know some other valid VCD use case). Maybe it's not the case here - I'm just double checking 🙂

Reply:
No, VCD didn't throw an error. It is just that you can't do it (unless you hack the UI I guess 😄 ). I checked what happened with Terraform/API, it seems it simply ignores the CPU/Memory and uses the Sizing Policy value, without any error (i.e.: if referenced `sizing_policy_id` has 1 CPUs and `vcd_vapp_vm` has `cpus = 2`, resulting VM has 1 CPU)

Maybe I should just remove this "smart" error checks and let this go through? It seems the result is the same, and hopefully the user will know that the reason is the Sizing Policy was applied. Also what you mentioned is quite relevant.

Reply:
Ah, the only issue is that this causes Terraform to think the VM needs an update everytime in subsequent applies:
```
resource "vcd_vapp_vm" "test-vm" {
        id                             = "urn:vcloud:vm:abc8cb18-8232-4401-b876-d3b1f5028056"
      ~ memory                         = 1024 -> 512
        name                           = "TestAccVcdVAppVm_SizingPolicy"
        # (31 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

Reply:
Hm, it still seems to me we should not throw a fatal error here, because VCD doesn't throw it. Here's one more thought: what about printing a warning instead?

Reply:
Sounds good! Changed.

Reply:
Thanks! CCing @Didainius for reviewing our decision too :)

Reply:
Great!